pkgs/applications/terminal-emulators/mlterm/default.nix

```
@@ -19,25 +79,42 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-gfs5cdwUUwSBWwJJSaxrQGWJvLkI27RMlk5QvDALEDg=";
   };
 
-  nativeBuildInputs = [ pkg-config autoconf wrapGAppsHook ];
+  nativeBuildInputs = [
+    pkg-config
+    autoconf
+  ] ++ lib.optionals (toolsChoices.mlconfig or false) [
+    wrapGAppsHook
+  ];
   buildInputs = [
-    libX11
+    gtk
+    vte
+  ] ++ lib.optionals (imagelibChoice == "gdk-pixbuf") [
     gdk-pixbuf.dev
-    cairo
+  ] ++ lib.optionals (typeEnginesChoices.xcore or false) [
```

Nit: I feel like this pattern should be extracted. Something like

```suggestion
  ] ++ lib.optionals (featureEnabled "xcore") [
```

would read nicely.

    featureEnabled = feature: enableFeatures.${feature} or false;

---

I think we can remove the `or false`. Now since all features are either explicitly enabled or disabled, a user that is missing an attribute will get an error, which is fine IMO - they should update the attribute set via the relevant `passthru` attribute.